### DISCUSSION TITLE
Create port on a non-svg element

### DISCUSSION THREAD

**Discussion by seekeramento** [clientIO/joint#1782] [Q&A]
What I am trying to do is to have a port on a div element so that I can draw a link/connector that starts from this div. When I move this div, the link source will move accordingly.

**Comment by kumilingus** [comment]
How did you create this HTML element? You can not connect to `HTMLElements` in JointJS. You have to draw an SVG port under the HTML port and let the HTML port propagate pointer events with `pointer-events: none`.

**Reply by seekeramento** [reply]
This is what I am trying to do at the moment. The main challenge is how to easily make the port inherit the position of this html node all the time or vice versa. 
For your question, it is a normal div that I appended to the paper container, and it takes its position from a rect element of the cell I am trying to have the ports on. 

Do you have examples that do something similar?

**Reply by kumilingus** [reply]
Here are examples of how to render HTML element views (none of them are rendering ports):
- **JS** - https://resources.jointjs.com/demos/tasks
- **VUE** - https://github.com/clientIO/joint/tree/master/demo/vuejs
- **REACT** -https://github.com/clientIO/joint-paper-react

**Reply by seekeramento** [reply]
Thanks.
I know that a Link's source can be a point (x, y). We can have a link source coming from a coordinate, while the target can be another cell. 
My question is: is there an example or a way that emulates starting a new link when clicking on an html node, so as we move we can connect this link to a target call, while the source will be the point of the html node?

**Reply by kumilingus** [reply]
You should check the [source code](https://github.com/clientIO/joint/blob/master/src/linkTools/Connect.mjs) of the `Connect` tool. It shows how to create a link from an `SVGElement` which is not part of the `dia.ElementView`.

And this is how to use the `connectionStrategy` to pin the source of the link to a point.
```js
paper.options.connectionStrategy = (end, a, b, c, d, endType) => {
  if (endType === 'source') return { x: 100, y: 100 };
  return end;
}
```

**Reply by seekeramento** [reply]
Will take a look. Thanks

**Reply by seekeramento** [reply]
What function/option should I use to add the new Connection once the mouseup is triggered?

**Reply by kumilingus** [reply]
The `dragstart`, `drag` and `dragend` are methods of your interest ([code](https://github.com/clientIO/joint/blob/master/src/linkTools/Connect.mjs#L60-L84)). `dragend` is supposed to be triggered on `mouseup`.
